### PR TITLE
Preserve data in browser

### PR DESCRIPTION
**Pull request by ONJoseph**
- add localstorage.js functionality
- add the localstorage.js script to the index.html file
- add browser query

### CODE REVIEW

File: src/localstorage.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            || e.code === 1014
+            // test name field too, because code might not be present
+            // everything except Firefox
+            || e.name === 'QuotaExceededError'
+            // Firefox
+            || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+            // acknowledge QuotaExceededError only if there's something already stored
+            && (storage && storage.length !== 0);
+  }
+}
```
Comment (smunozmo) [inline src/localstorage.js]: - [ ] You are almost there! But we still need to use `localStorage` to meet the project requirements: 

You should implement the following interactions:

- When the user changes the content of any input field, the data is saved to the local storage.

- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

You should use the following data model:

- You must create a single JavaScript object with all the data from the entire form and save it in local storage. 
- Do not create one local storage entry per input field.

**Review by smunozmo — CHANGES_REQUESTED** [review]
Hi @ONJoseph, 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights 👏 

- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 🤓 

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

⚠️  **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ONJoseph** [comment]
Thanks a lot👍👍@*smunozmo *
For the encouraging feedback. Will try my best to implement the changes
requested. Please, any helpful pointers will be appreciated.

**Comment by smunozmo** [comment]
@ONJoseph Sure! The main idea of the project is to get the user data from the form and store it as an object in the `localStorage` so that when you reload the page, the form will be pre-filled with this data.

**Comment by ONJoseph** [comment]
> @ONJoseph Sure! The main idea of the project is to get the user data from the form and store it as an object in the `localStorage` so that when you reload the page, the form will be pre-filled with this data.

Hi @smunozmo, Have struggled with it and implemented as much as I can now. Have resubmitted for approval. Will continue to improve my portfolio as I progress. Thanks a lot!

**Review by aimemalaika — APPROVED** [review]
Hi @ONJoseph , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

133591850-fdcb7dd3-dd7a-4274-81e7-0e59434148f7

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ONJoseph** [comment]
I'm super grateful for the helpful review and guidance👍👍@*aimemalaika*
*Cheers 🥂🥂*

**Pull request merged**